packages/crayons-core/src/components/radio/radio.tsx

```diff
@@ -100,25 +103,28 @@ export class Radio {
       <Host
         class='radio-container'
         onClick={() => this.toggle()}
-        role='radio'
-        tabIndex='0'
+        role="radio"
+        tabIndex="-1"
+        aria-label={this.label}
+        aria-describedby="description"
         aria-disabled={this.disabled ? 'true' : 'false'}
-        aria-checked={this.checked}
+        aria-checked={`${this.checked}`}
         onFocus={() => this.onFocus()}
         onBlur={() => this.onBlur()}
-      >
-        <input type='radio' ref={(el) => (this.radio = el)}></input>
-        <label>
-          <span class='text'>
-            <slot />
-          </span>
-          <br />
-          {this.label !== '' ? (
-            <span class='label-field'>{this.label}</span>
-          ) : (
-            ''
-          )}
-        </label>
+        >
+          <input type="radio"
+            ref={el => this.radio = el}>
+          </input>
+          <label>
+          {
+              this.label !== ''
+              ? <span class="label-field">{this.label}</span>
```

Comment:
Same comment as in `checkbox`. Possible to render as text node (without the `span`) and style this using the `label` tag? (assuming we're adding class only for styling)

Reply:
We are using `label-field` class for styling only. I didn't modify since it was already implemented that way. We can change it to as suggested. Have modified the same in the latest commit